jupyter-extension/src/plugins/mount/mountDrive.ts

```diff
@@ -30,32 +55,61 @@ export class MountDrive implements Contents.IDrive {
     return this._isDisposed;
   }
 
+  async _get(
+    url: string,
+    options: PartialJSONObject,
+  ): Promise<Contents.IModel> {
+    url += URLExt.objectToQueryString(options);
+    return requestAPI<Contents.IModel>(url, 'GET');
+  }
+
+  // We might be able to pass pagination to the ContentsManager backend by including
+  // it in the URI specifier. This would require a custom ContentsManager which
+  // supports pagination.
   async get(
     localPath: string,
     options?: Contents.IFetchOptions,
   ): Promise<Contents.IModel> {
+    const url = URLExt.join('pfs', localPath);
+
+    // Make a no-content request to determine the type of content being requested.
+    // If the content type is not a directory, then we want to preserve our cache
+    //   and not paginate the results.
+    let shallowResponse;
     try {
-      const response = await requestAPI<Contents.IModel>(
-        URLExt.join('pfs', localPath),
-        'GET',
-      );
-      return response;
+      shallowResponse = await this._get(url, {content: '0'});
     } catch (e) {
       console.log('/pfs not found');
```

Comment:
I don't think this log message is always true and I don't think we should return a default content model. Instead we should do some sort of initialization check to see if the mount point exists and throw errors as normal.

Reply:
[ticket](https://pachyderm.atlassian.net/browse/INT-1128?atlOrigin=eyJpIjoiYzVhZjY1ZWVlNzA4NDdmYjgwZWQ5YWE4ODg4YzIzMTEiLCJwIjoiaiJ9)